.github/workflows/screener-build.yml

```diff
@@ -28,8 +37,8 @@ jobs:
           DEPLOYHOST='fluentuipr.z22.web.core.windows.net'
           BUILD_BUILDID=${{ github.run_id }}
           BUILD_SOURCEBRANCH=${{ github.ref }}
```

Comment:
I am a bit confused with this, we have two places where it's declared now: there and line 10:

![image](https://user-images.githubusercontent.com/14183168/186367366-e60db3b6-adda-463a-a94d-839cc9e3ebc4.png)

Is it intentional?

Reply:
the second declaration is actually creating a file that will be uploaded as an artifact so that the second workflow will have access to the same env variables, so the first declaration is to have the env variables in the current workflow, and the second one is to send them to the `screener-run.yml` workflow

Reply:
The problem that I see is that we do the same in different areas: we create an artifact for `screener-run.yml` & we define the same for current workflow.

Can we somehow reuse `artifacts/environment` file in `screener-react-northstar` and others? For example:


```diff
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0

-      - name: Set env variables if there is a PR
-        run: |
-          echo "SYSTEM_PULLREQUEST_TARGETBRANCH=${{ github.event.pull_request.base.ref }}" >> $GITHUB_ENV
-          echo "SYSTEM_PULLREQUEST_SOURCEBRANCH=${{ github.event.pull_request.head.ref }}" >> $GITHUB_ENV
-          echo "SYSTEM_PULLREQUEST_PULLREQUESTID=${{ github.event.pull_request.id }}" >> $GITHUB_ENV
-          echo "SYSTEM_PULLREQUEST_SOURCECOMMITID=${{ github.event.pull_request.head.sha }}" >> $GITHUB_ENV
-          echo "ISPR=true" >> $GITHUB_ENV
-          echo "DEPLOYBASEPATH=pull/$PR_NUMBER" >> $GITHUB_ENV
-          echo "DEPLOYURL=https://$DEPLOYHOST/pull/$PR_NUMBER" >> $GITHUB_ENV
-          echo "BUILD_SOURCEBRANCHNAME='merge'" >> $GITHUB_ENV
-        env:
-          DEPLOYHOST: 'fluentuipr.z22.web.core.windows.net'
-          PR_NUMBER: ${{ github.event.number }}
-        if: ${{startsWith(github.ref, 'refs/pull/')}}

-      - name: Set env variables if there is not a PR
-        run: |
-          echo "ISPR=false" >> $GITHUB_ENV
-          echo "DEPLOYBASEPATH=heads/$SYSTEM_PULLREQUEST_TARGETBRANCH" >> $GITHUB_ENV
-          echo "DEPLOYURL=https://$DEPLOYHOST/heads/$SYSTEM_PULLREQUEST_TARGETBRANCH" >> $GITHUB_ENV
-          echo "BUILD_SOURCEBRANCHNAME=${{ github.ref_name }}" >> $GITHUB_ENV
-        if: ${{!startsWith(github.ref, 'refs/pull/')}}

+- name: Download a single artifact
+  uses: actions/download-artifact@v3
+  with:
+    name: env-artifact

+      - name: Define env variables
+        run: |
+          input_file="environment"
+          while read line
+          do
+          echo "$line" >> $GITHUB_ENV
+          done < "$input_file"

      - uses: actions/setup-node@v3
        with:
          node-version: 14.18.1
          cache: 'yarn'
```

Reply:
yes, good point- since I am already creating the artifact, I can also use it within the same workflow